### ISSUE TITLE
Notifications are effectively useless

### ISSUE THREAD

**Issue by halcyon1234** [apxltd/what-bugs#33]
File this under Bikeshed if you want, but I'm finding the Notifications box pretty much useless.  I click on it, and I see three notifications that give me very little information about what they are, or why I should care.

Example:
http://i.imgur.com/VBfEh7m.png

There's 3 of 36 notifications being displayed.  Yes, there's a scrollbar but-- so what?  A flyout like that is meant to pop open, look at, interact with a single click and/or close.

The "Notifications" tooltip actually covers up one of the pieces of information (the time of the first notification).  So I'll need to move the mouse and wait for the tooltip to go away-- meaning something that is supposed to be fast and with minimal UX is now slower and with an extra UX step.

Then there's the content itself.  What am I being notified of?  Well, I could read all three of those and find out-- but there's another issue with the information layout of the notifications.  This is supposed to be a quick-access popup.  I should get a better sense of what the notifications are WITHOUT having to read them.  I need to hunt for the information "upvoted" and "mentioned".  And the problems there are:

1)  both those keywords are normal weight text, making them hard to find, especially in a sea of attention-grabbing BOLD text.
2) all those keywords are sprinkled about in different positions.  Upvoted is word 3, middle of first line.  Mentioned is second word first line, mostly left aligned.  Upvoted (again) is in the middle of a paragraph.

Then what is that empty circle?  Is that online notification?  What about the third notification-- is that Ned's status, or one of the 2 others?  Their status at time of notifiying, or now?  And who cares?

And finally-- what are those notifications regarding?  A post in a thread.  Oh boy.  Which one?  

As such, the Notifications flyout is effectively useless.  There is too little information, and what little there is is so haphazard that it is a chore to determine what the information is.

Suggested alternate layout.  Column-like data, so that related pieces of information are aligned with each other.  The eye needs to be able to just flow down the list without jerking back and forth.

Column 1:  Notification icon. There's got to be icons for Upvote, Reply, Mention, etc.  FFS, if Dishcourse can have meaningful icons...

Column 2: Avatar & username.  I suppose "+2 others" is okay.  It's pretty useless, still.  Then again, having a butt-ton of notifications for the same post / chat overload the dropdown, so it's a lose/lose there

Column 3: Information about what is being notified.  Top line, thread/chat title.  Second line: the first few non-quoted words of the post.  That way if someone likes my post in a certain thread, I at least get a hint as to which post it is-- so I don't have to click into the post to see it.

Column 4: The "time ago" shit.  Drop the online status. No one cares.

Also suggested:
- smaller font
- taller box
- wider box

All three of those will allow for more than 3 notifications to be seen at once.  I'd hazard a guess that any less than 10 notifications at a time make the box useless.

**Comment by LB--** [comment]
Yeah, as buggy as it was I'd rather take the Discourse notification tray over the current NodeBB one.

**Comment by LB--** [comment]
Just here to say this is still an issue. Notifications for watched topics generally link to the most recent unread post instead of the earliest, and even the number of new posts is wrong - it may say there are two new posts but there will actually be 10 and it only was aware of the most recent two. Generally I just scroll up until I see a post I recognize or have upvoted and read from there.

**Comment by boomzillawtf** [comment]
I believe this has been reported upstream:

https://github.com/NodeBB/NodeBB/issues/4469
